.travis.yml

```diff
@@ -123,7 +124,7 @@ script:
   - if $INSTALLED; then echo cabal new-build -w ${HC} --disable-tests --disable-benchmarks $(${HCPKG} list --global --simple-output --names-only | sed 's/\([a-zA-Z0-9-]\{1,\}\) */--constraint="\1 installed" /g') all | sh; else echo "Not building with installed constraints"; fi
 
   # build & run tests, build benchmarks
-  - cabal new-build -w ${HC} ${TEST} ${BENCH} -j2 all
+  - cabal new-build -w ${HC} ${CABALFLAGS} ${TEST} ${BENCH} -j2 all
```

Comment:
You'll need to pass `${CABALFLAGS}` to more than just here. You'll also need it on lines 121, 124, 128, and 137, which call `new-build`, `new-test`, and `new-haddock`. (Otherwise, when those commands are run, it will first calculate a new set of dependencies, since you passed a different set of flags!)

Reply:
Ah, thanks. I had forgotten that `new-test` even existed, so I didn't think to search for that. Also, it's a little strange to me that this to the earlier `new-build` commands in the file (the ones that don't build suite) would have any affect at all since `primitive` doesn't actually depend on `quickcheck-classes`.

Reply:
To be clear, it's not necessarily guaranteed that not passing `${CABALFLAGS}` would make a difference. But in my experience, things tend to work best when you use a consistent set of build flags everywhere—you don't want to give the recompilation checker any excuse for believing that some dependency needs to be rebuilt.